aws-rds-dbsecuritygroup/src/main/java/software/amazon/rds/dbsecuritygroup/BaseHandlerStd.java

```diff
@@ -0,0 +1,264 @@
+            .delay(Duration.ofSeconds(30L)).build();
+
+    protected static boolean compareEC2SecurityGroupToIngres(final EC2SecurityGroup group, final Ingres ingres) {
+        return compareStrings(group.ec2SecurityGroupId(), ingres.getEC2SecurityGroupId()) &&
+                compareStrings(group.ec2SecurityGroupName(), ingres.getEC2SecurityGroupName()) &&
+                compareStrings(group.ec2SecurityGroupOwnerId(), ingres.getEC2SecurityGroupOwnerId());
+    }
+
+    private static boolean compareStrings(String str1, String str2) {
+        return (str1 == null ? str2 == null : str1.equals(str2));
+    }
+
+    @Override
+    public final ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger) {
+        return handleRequest(
+                proxy,
+                request,
+                callbackContext != null ? callbackContext : new CallbackContext(),
+                proxy.newProxy(ClientBuilder::getClient),
+                logger
+        );
+    }
+
+    protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<RdsClient> proxyClient,
+            final Logger logger);
+
+    protected ProgressEvent<ResourceModel, CallbackContext> updateIngresses(
+            final AmazonWebServicesClientProxy proxy,
+            final ProxyClient<RdsClient> proxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final ResourceModel model,
+            final Collection<Ingres> desiredIngresses
+    ) {
+        if (progress.getCallbackContext().isIngresModified()) {
+            return progress;
+        }
+
+        final Optional<DBSecurityGroup> dbSecurityGroup = proxyClient.injectCredentialsAndInvokeV2(
+                Translator.describeDBSecurityGroupsRequest(model),
+                proxyClient.client()::describeDBSecurityGroups
+        ).dbSecurityGroups().stream().findFirst();
+
+        if (!dbSecurityGroup.isPresent()) {
+            throw new CfnNotFoundException(ResourceModel.TYPE_NAME, model.getGroupName());
+        }
+
+        final Collection<Ingres> currentIngresses = dbSecurityGroup.get()
+                .ec2SecurityGroups()
+                .stream()
+                .map(Translator::translateDBSecurityGroupIngresFromSdk)
+                .collect(Collectors.toSet());
+
+        return progress
+                .then(p -> revokeIngresses(proxy, proxyClient, p, currentIngresses))
+                .then(p -> authorizeIngresses(proxy, proxyClient, p, desiredIngresses));
```

Comment:
This is suboptimal: we should find diff before triggering this update as it is done in the tag handler